### PR TITLE
Support for /vrp-long with new Job model

### PR DESCRIPTION
**Pull request by rogertsui0923**

### CODE REVIEW

**Comment by rogertsui0923** [comment]
@oddg Ready for review and perhaps suggestions for additional test cases?

**Comment by rogertsui0923** [comment]
@oddg do you think there is a good reason to support both `Routific.getRoute` and `Routific.new.getRoute`? I had implemented the async `get_route_async` only as an instance method, but if we decide to support both, I can refactor the job to call the class itself for updates (then the job won't have to keep a Routific instance).

**Comment by rogertsui0923** [comment]
@oddg, I have moved the fetch job call to the Job model, ready for review! I will make a separate PR to remove the class methods and to fix how the token is set (I will also move the call to `validate_and_prefix_token` when it is set and not when we are about to make the http request).

File: spec/routific_spec.rb [hunk 1]
```diff
@@ -125,6 +129,24 @@
         expect(route).to be_instance_of(RoutificApi::Route)
       end
     end
+
+    describe "#get_route_async" do
+      before do
+        job = nil
+        set_visit_and_vehicle routific
```
Comment (oddg) [inline spec/routific_spec.rb]: Should we clear the existing visits and vehicle first?

Comment (oddg) [reply spec/routific_spec.rb]: nevermind, i just read you comment....

Comment (oddg) [reply spec/routific_spec.rb]: It seems that the :visits and :fleet attributes are cleared by the `describe`. So that's fine.

**Pull request merged**